main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripAndShapeDistanceValidator.java

```diff
@@ -32,55 +35,75 @@
-
+  private final GtfsStopTableContainer stopTable;
   private final GtfsShapeTableContainer shapeTable;
+  private final double DISTANCE_THRESHOLD = 11.1; // distance in meters
 
   @Inject
   TripAndShapeDistanceValidator(
       GtfsTripTableContainer tripTable,
       GtfsStopTimeTableContainer stopTimeTable,
+      GtfsStopTableContainer stopTable,
       GtfsShapeTableContainer shapeTable) {
     this.tripTable = tripTable;
     this.stopTimeTable = stopTimeTable;
     this.shapeTable = shapeTable;
+    this.stopTable = stopTable;
   }
 
   @Override
   public void validate(NoticeContainer noticeContainer) {
-    shapeTable
-        .byShapeIdMap()
+    tripTable
+        .getEntities()
         .forEach(
-            (shapeId, shape) -> {
-              double maxShapeDist =
+            trip -> {
+              String shapeId = trip.shapeId();
+              // Get distance for trip
+              int nbStopTimes = stopTimeTable.byTripId(trip.tripId()).size();
+              if (nbStopTimes == 0) {
+                return;
+              }
+              GtfsStopTime lastStopTime =
+                  stopTimeTable.byTripId(trip.tripId()).get(nbStopTimes - 1);
+              GtfsStop stop = stopTable.byStopId(lastStopTime.stopId()).orElse(null);
+              if (stop == null) {
+                return;
+              }
+              double maxStopTimeDist = lastStopTime.shapeDistTraveled();
+
+              // Get max shape distance for trip
+              GtfsShape maxShape =
                   shapeTable.byShapeId(shapeId).stream()
-                      .mapToDouble(GtfsShape::shapeDistTraveled)
-                      .max()
-                      .orElse(Double.NEGATIVE_INFINITY);
-
-              tripTable
-                  .byShapeId(shapeId)
-                  .forEach(
-                      trip -> {
-                        double maxStopTimeDist =
-                            stopTimeTable.byTripId(trip.tripId()).stream()
-                                .mapToDouble(GtfsStopTime::shapeDistTraveled)
-                                .max()
-                                .orElse(Double.NEGATIVE_INFINITY);
-
-                        if (maxStopTimeDist > maxShapeDist) {
-                          noticeContainer.addValidationNotice(
-                              new TripDistanceExceedsShapeDistanceNotice(
-                                  trip.tripId(), shapeId, maxStopTimeDist, maxShapeDist));
-                        }
-                      });
+                      .max(Comparator.comparingDouble(GtfsShape::shapeDistTraveled))
+                      .orElse(null);
+              if (maxShape == null) {
+                return;
+              }
+
+              double maxShapeDist = maxShape.shapeDistTraveled();
+              double distanceInMeters =
+                  getDistanceMeters(maxShape.shapePtLatLon(), stop.stopLatLon());
+              if (maxStopTimeDist > maxShapeDist) {
```

Comment:
I must admin I am a bit confused. You could have cases where maxStopTimeDist is way bigger than maxShapeDist, but where the coordinates are the same, and this would not generate an error?
@emmambd

Reply:
@jcpitre Discussed in meeting - good point! We talked about the notice defaulting to returning a warning in this case, rather than returning nothing. cc @cka-y for thoughts

Reply:
I agree it's a bit confusing. We've established a threshold for latitude/longitude coordinates, but our previous analysis shows that it doesn't always logically correspond to the distance traveled. Given we do not infer units, I believe the best we could do, which aligns with our existing default behavior, is to continue issuing a warning.

Reply:
Agreed.

Reply:
Sorry I think I was not clear enough when I wrote:
> You could have cases where maxStopTimeDist is way bigger than maxShapeDist, but where the coordinates are the same, and this would not generate an error?

It should have ended with: `and this would generate neither error nor warning`

It still seems odd that the situation where maxStopTimeDist > maxShapeDist would not be reported at all.
It is a corner case (coordinates have to be the same but distances not the same) but I think it's also a low hanging fruit.

Reply:
just pushed a fix to address your concerns @jcpitre